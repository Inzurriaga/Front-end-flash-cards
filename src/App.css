html, body {
  height: 100%;
  overflow: hidden;
}

h1 {
  text-align: center;
  margin: 0;
}

.nav {
  position: fixed;
  bottom: 0;
  left: 50px;
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.nav li {
  font-size: 1.5em;
  display: list-item;
  height: 60px;
  margin-bottom: 30px;
  transform: translateX(-300px);
  transition: transform .7s 0s cubic-bezier(1.000, 0.005, 0.240, 1.000);
}

.nav li {
  display: block;
  transition: color 0.7s;
}

#root {
  height: 100%;
}
.page {
  height: 100%;
  background-color: pink;
}

.main-content {
  height: 200%;
  position: relative;
  transform-origin: top left;
  transition: transform 0.7s cubic-bezier(1, 0.005, 0.24, 1), -webkit-transform 0.7s cubic-bezier(1, 0.005, 0.24, 1);
  background-color: blue;
}

.hamburger-container {
  z-index: 1;
  position: fixed;
  top: 0;
  left: 0;
  height: 150px;
  width: 150px;
  cursor: pointer;
}

.hamburger-circle {
  position: absolute;
  z-index: -1;
  transition: 0.4s;
  height: 0;
  width: 0;
  background-color: yellow;
  border-bottom-right-radius: 100%;
}

.hamburger-container:hover .hamburger-circle{
  height: 150px;
  width: 150px;
  background-color: yellow;
  border-bottom-right-radius: 100%;
}

.what .main-content {
  transform: rotate(-30deg)
}

.what li {
  transform: translateX(0);
  transition: transform 0.7s .46s cubic-bezier(0.175, 0.885, 0.320, 1.275);
}


.what li:nth-child(2){
   transition-delay: .47s;
  }
.what li:nth-child(3){
    transition-delay: .48s;
  }
.what li:nth-child(4){
    transition-delay: .49s;
  }
.what li:nth-child(5){
    transition-delay: .50s;
  }
.what li:nth-child(6){
    transition-delay: .51s;
  }


  /* the hamburger */

  .stick-container {
    position: absolute;
    top: 25%;
    left: 25%;
    transform: translate( -25%, -25%)
  }

  .stick {
    height: 10px;
    width: 75px;
    background-color: crimson;
    border-radius: 25px;
    margin: 5px;
  }

 .stick-one.open{
  animation: open-stick-one 0.5s forwards;
}

.stick-two.open  {
  animation: open-stick-two 0.25s forwards;
}

.stick-three.open  {
  animation: open-stick-three 0.5s forwards;
}

.stick-one.close {
  animation: close-stick-one 0.5s forwards;
}

.stick-two.close {
  animation: close-stick-two 0.5s forwards;
}

.stick-three.close {
  animation: close-stick-three 0.5s forwards;
}

@keyframes open-stick-one {
  0% {
    width: 75px;
  }
  50% {
    width: 10px;
    transform: translate( 0, 0)
  }
  75%, 76% {
    width: 10px;
    transform: translate( 0, 15px)
  }
  77% {
    transform: rotate(-45deg) translate( 0, 34px);
  }
  100%{
    transform: rotate(45deg) translate( 0px, 34px);
    width: 75px;
  }
}

@keyframes open-stick-two {
  0% {
    width: 75px;
  }
  100% {
    width: 10px;
  }
}

@keyframes open-stick-three {
  0% {
    width: 75px;
  }
  50% {
    width: 10px;
    transform: translate( 0, 0)
  }
  75%, 76% {
    width: 10px;
    transform: translate( 0, -15px)
  }
  77% {
    transform: rotate(-45deg) translate( 0, -34px);
  }
  100%{
    transform: rotate(-45deg) translate( 0, -34px);
    width: 75px;
  }
}

@keyframes close-stick-one {
  0% {
    width: 10px;
    transform: translate( 0, 15px)
  }
  50% {
    width: 10px;
    transform: translate( 0, 0)
  }
  100% {
    width: 75px;
  }
}

@keyframes close-stick-two {
  0% {
    width: 10px;
  }
  50% {
    width: 10px;
  }
  100% {
    width: 75px;
  }
}

@keyframes close-stick-three {
  0% {
    width: 10px;
    transform: translate( 0, -15px);
  }
  50% {
    width: 10px;
    transform: translate( 0, 0)
  }
  100% {
    width: 75px;
  }
}
