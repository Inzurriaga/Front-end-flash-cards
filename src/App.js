import React, { Component } from 'react';
import MainMenu from "./MainMenu";
import Nav from "./NavBar"
import ListComponent from "./listcomponent"
import './App.css';
import './normalize.css'

class App extends Component {
  constructor() {
    super();
    this.state = {
        cards: [{
          catagory: "html",
          question: "What are the three main tags that all HTMl structure needs?",
          answer: "<html>, <head>, and <body>",
          falseAnswer: ["<header>, <section>, and <footer>", "<div>, <h1>, and <p>", "<section>, <div>, and <span>"],
          color: "#f1652a"
        }, {
          catagory: "css",
          question: "Which property allows you to customize your cursor?",
          answer: "cursor",
          falseAnswer: ["point","cursor-style","mouse"],
          color: "#2fa4d5"
        }, {
          catagory: "javascript",
          question: "Which method allows you to remove a datatype at the end of an array?",
          answer: "arrayEx.pop(insertDatatype)",
          falseAnswer: ["arrayEx.push(insertDatatype)","arrayEx.remove(insertDatatype)","arrayEx.delete(insertDatatype)"],
          color: "#f7d33c"
        }],
        page: "main menu"
    }
  }

  toggleMenu() {
    let page = document.querySelector(".page");
    let sticks = document.querySelectorAll(".stick")
      if(!page.classList.contains("what")){
        page.classList.add("what");
      }else{
        page.classList.remove("what");
      }
      sticks.forEach((stick) => {
      if(!stick.classList.contains("open") || stick.classList.contains("close"))  {
        stick.classList.remove("close");
        stick.classList.add("open")
      } else{
        stick.classList.remove("open");
        stick.classList.add("close")
      }
    })
  }

  togglecomponents(string) {
    this.setState({ page: string })
  }

  render() {
    if(this.state.page === "main menu") {
        return ( 
        <div className="page">
          <Nav toggleMenu={this.toggleMenu}  />
          <section className="main-content">
            <header>
              <h1>flash card for coding</h1>
            </header>
            <MainMenu />
          </section>
        </div>
      );
        }else {
          return ( 
            <div className="page">
              <Nav toggleMenu={this.toggleMenu}
                toggleComponents={this.togglecomponents}/>
              <section className="main-content">
                <header>
                  <h1>flash card for coding</h1>
                </header>
                <ListComponent cards={this.state.cards}/>
              </section>
            </div>
          );
        }
}
}

export default App;
