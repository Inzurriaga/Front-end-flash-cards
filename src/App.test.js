import React from 'react';
import App from './App.js';
import { shallow } from 'enzyme';

const toggle = jest.fn();

describe('app', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />);
  });

  it('should display app', () => {
    expect(wrapper).toMatchSnapshot();
  });
});