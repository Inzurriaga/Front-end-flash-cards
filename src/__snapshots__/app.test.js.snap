// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should display app 1`] = `
<div
  className="page"
>
  <Nav
    toggleComponents={[Function]}
    toggleMenu={[Function]}
  />
  <section
    className="main-content"
  >
    <header>
      <h1
        onClick={[Function]}
      >
        Flash Cards For Dummies
      </h1>
    </header>
    <MainMenu
      toggleComponents={[Function]}
    />
  </section>
</div>
`;

exports[`teaList should display all tea cards 1`] = `
<div
  className="page"
>
  <Nav
    toggleComponents={[Function]}
    toggleMenu={[Function]}
  />
  <section
    className="main-content"
  >
    <header>
      <h1
        onClick={[Function]}
      >
        Flash Cards For Dummies
      </h1>
    </header>
    <MainMenu
      toggleComponents={[Function]}
    />
  </section>
</div>
`;

exports[`teaList should display app 1`] = `
<div
  className="page"
>
  <Nav
    toggleComponents={[Function]}
    toggleMenu={[Function]}
  />
  <section
    className="main-content"
  >
    <header>
      <h1
        onClick={[Function]}
      >
        Flash Cards For Dummies
      </h1>
    </header>
    <MainMenu
      toggleComponents={[Function]}
    />
  </section>
</div>
`;
